main.go

```diff
@@ -66,11 +67,15 @@ func main() {
 		log.Fatalf("failed to create metadata: %v", err)
 	}
 	// Memcache
-	branchCacheFolder, err := server.ReadBranchCacheFolder(
-		ctx, branchCacheBucket, branchCacheVersionFile)
-	if err != nil {
-		log.Fatalf("failed to read branch cache folder: %v", err)
+	branchCacheFolder := *branchFolder
+	if branchCacheFolder == "" {
+		branchCacheFolder, err = server.ReadBranchCacheFolder(
+			ctx, branchCacheBucket, branchCacheVersionFile)
```

Comment:
Once we launch its going to be very tough to add anything to branch cache without risk of breaking prod. Not this PR, but soon, should we make separate paths for prod vs. staging and separate pubsub topics? Such that, the downstream pipeline writes to staging, and after validation, we'll have a script to: 1. copy over the branch cache folder file, 2. send prod notification.

Reply:
yea, that sounds reasonable.